subsys/net/lib/app/client.c

```diff
@@ -322,7 +322,20 @@ int net_app_init_client(struct net_app_ctx *ctx,
 	if (client_addr) {
 		memcpy(&addr, client_addr, sizeof(addr));
 	} else {
-		addr.sa_family = AF_UNSPEC;
+		if (peer_addr_str) {
```

Comment:
There's also "peer_addr" arg to this function (net_app_init_client), shouldn't it be consulted to avoid extra name resolution run?

Also, later we'd need to resolve peer_addr_str anyway, so make sense to do that once, though looking at the code, that would require heavier refactoring of the code...